pepdbagent/modules/view.py

```diff
@@ -0,0 +1,405 @@
+        sub_project_dict = {"_config": config, "_sample_dict": samples, "_subsample_dict": None}
+        if raw:
+            return sub_project_dict
+        else:
+            return peppy.Project.from_dict(sub_project_dict)
+
+    def get_annotation(
+        self, namespace: str, name: str, tag: str = DEFAULT_TAG, view_name: str = None
+    ) -> ViewAnnotation:
+        """
+        Get annotation of the view.
+
+        :param namespace: namespace of the project
+        :param name: name of the project
+        :param tag: tag of the project
+        :param view_name: name of the sample
+        :return: ViewAnnotation object:
+            {project_namespace: str,
+             project_name: str,
+             project_tag: str,
+             name: str,
+             description: str,
+             number_of_samples: int}
+        """
+        view_statement = select(Views).where(
+            and_(
+                Views.project_mapping.has(namespace=namespace, name=name, tag=tag),
+                Views.name == view_name,
+            )
+        )
+
+        with Session(self._sa_engine) as sa_session:
+            view = sa_session.scalar(view_statement)
+            if not view:
+                raise ViewNotFoundError(
+                    f"View {name} of the project {namespace}/{name}:{tag} does not exist"
+                )
+            return ViewAnnotation(
+                project_namespace=namespace,
+                project_name=name,
+                project_tag=tag,
+                name=view.name,
+                description=view.description,
+                number_of_samples=len(view.samples),
+            )
+
+    def create(
+        self,
+        view_name: str,
+        view_dict: Union[dict, CreateViewDictModel],
+        description: str = None,
+    ) -> None:
+        """
+        Create a view of the project in the database.
+
+        :param view_name: namespace of the project
+        :param view_dict: dict or CreateViewDictModel object with view samples.
+            Dict should have the following structure:
+                {
+                    project_namespace: str
+                    project_name: str
+                    project_tag: str
+                    sample_list: List[str] # list of sample names
+                }
+        :param description: description of the view
+        retrun: None
+        """
+        if isinstance(view_dict, dict):
+            view_dict = CreateViewDictModel(**view_dict)
+
+        project_statement = select(Projects).where(
+            and_(
+                Projects.namespace == view_dict.project_namespace,
+                Projects.name == view_dict.project_name,
+                Projects.tag == view_dict.project_tag,
+            )
+        )
+
+        with Session(self._sa_engine) as sa_session:
+            project = sa_session.scalar(project_statement)
+            if not project:
+                raise ProjectNotFoundError(
```

Comment:
Why are we looking for existing projects if this is the `create()` function?

Reply:
We are creating view, not project